samples/ask-akka-agent/src/main/java/akka/ask/agent/application/Knowledge.java

```diff
@@ -0,0 +1,42 @@
+package akka.ask.agent.application;
+
+import akka.ask.common.MongoDbUtils;
+import akka.ask.common.OpenAiUtils;
+import com.mongodb.client.MongoClient;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.rag.AugmentationRequest;
+import dev.langchain4j.rag.DefaultRetrievalAugmentor;
+import dev.langchain4j.rag.RetrievalAugmentor;
+import dev.langchain4j.rag.content.injector.ContentInjector;
+import dev.langchain4j.rag.content.injector.DefaultContentInjector;
+import dev.langchain4j.rag.content.retriever.EmbeddingStoreContentRetriever;
+import dev.langchain4j.rag.query.Metadata;
+
+public class Knowledge {
+  private final RetrievalAugmentor retrievalAugmentor;
+  private final ContentInjector contentInjector = new DefaultContentInjector();
+
+  public Knowledge(MongoClient mongoClient) {
+    var contentRetriever = EmbeddingStoreContentRetriever.builder()
+        .embeddingStore(MongoDbUtils.embeddingStore(mongoClient))
+        .embeddingModel(OpenAiUtils.embeddingModel())
+        .maxResults(10)
+        .minScore(0.1)
+        .build();
+
+    this.retrievalAugmentor = DefaultRetrievalAugmentor.builder()
+        .contentRetriever(contentRetriever)
+        .build();
+  }
+
+  public String addKnowledge(String question) {
+    var chatMessage = new UserMessage(question);
+    var metadata = Metadata.from(chatMessage, null, null);
+    var augmentationRequest = new AugmentationRequest(chatMessage, metadata);
+
+    var result = retrievalAugmentor.augment(augmentationRequest);
+    UserMessage augmented = (UserMessage) contentInjector.inject(result.contents(), chatMessage);
```

Comment:
What this does is to construct a new user message with the template:
```
                    {{userMessage}}

                    Answer using the following information:
                    {{contents}}
```

This is the same as what is done by Langchain4j DefaultAiServices.